delta-lake/src/main/java/org/apache/iceberg/delta/BaseMigrateDeltaLakeTableAction.java

```diff
@@ -0,0 +1,333 @@
+      throw new IllegalArgumentException(
+          String.format("File %s does not specify a partitionValues", fullFilePath));
+    }
+
+    FileFormat format = determineFileFormatFromPath(fullFilePath);
+    InputFile file = hadoopFileIO.newInputFile(fullFilePath);
+    MetricsConfig metricsConfig = MetricsConfig.forTable(table);
+    String nameMappingString = table.properties().get(TableProperties.DEFAULT_NAME_MAPPING);
+    NameMapping nameMapping =
+        nameMappingString != null ? NameMappingParser.fromJson(nameMappingString) : null;
+    Metrics metrics = getMetricsForFile(file, format, metricsConfig, nameMapping);
+
+    String partition =
+        spec.fields().stream()
+            .map(PartitionField::name)
+            .map(name -> String.format("%s=%s", name, partitionValues.get(name)))
+            .collect(Collectors.joining("/"));
+
+    return DataFiles.builder(spec)
+        .withPath(fullFilePath)
+        .withFormat(format)
+        .withFileSizeInBytes(file.getLength())
+        .withMetrics(metrics)
+        .withPartitionPath(partition)
+        .build();
+  }
+
+  private FileFormat determineFileFormatFromPath(String path) {
+    if (path.endsWith(PARQUET_SUFFIX)) {
+      return FileFormat.PARQUET;
+    } else if (path.endsWith(AVRO_SUFFIX)) {
+      return FileFormat.AVRO;
+    } else if (path.endsWith(ORC_SUFFIX)) {
+      return FileFormat.ORC;
+    } else {
+      throw new ValidationException("The format of the file %s is unsupported", path);
+    }
+  }
+
+  private Metrics getMetricsForFile(
+      InputFile file, FileFormat format, MetricsConfig metricsSpec, NameMapping mapping) {
+    switch (format) {
+      case AVRO:
+        long rowCount = Avro.rowCount(file);
+        return new Metrics(rowCount, null, null, null, null);
+      case PARQUET:
+        return ParquetUtil.fileMetrics(file, metricsSpec, mapping);
+      case ORC:
+        return OrcMetrics.fromInputFile(file, metricsSpec, mapping);
+      default:
+        throw new ValidationException("Unsupported file format: %s", format);
+    }
+  }
+
+  private Map<String, String> destTableProperties(
+      io.delta.standalone.Snapshot deltaSnapshot, String originalLocation) {
+    additionalPropertiesBuilder.putAll(deltaSnapshot.getMetadata().getConfiguration());
+    additionalPropertiesBuilder.putAll(
+        ImmutableMap.of(
+            MIGRATION_SOURCE_PROP, DELTA_SOURCE_VALUE, ORIGINAL_LOCATION_PROP, originalLocation));
+
+    return additionalPropertiesBuilder.build();
+  }
+
+  /**
+   * Get the full file path, the input {@code String} path can be either a relative path or an
+   * absolute path of a data file in delta table
+   *
+   * @param path the return value of {@link AddFile#getPath()} or {@link RemoveFile#getPath()}
+   *     (either absolute or relative)
+   * @param tableRoot the root path of the delta table
+   */
+  @VisibleForTesting
```

Comment:
This method is simple enough that I don't think we need to unit test it and mark it as VisibleForTesting, what do you think?